File: SQL Server 2012/Projects/C#/Chapter 20a/MusicStoreImageManager/ImageManagerForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;

using System.IO;

namespace MusicStoreImageManager
{
    public partial class ImageManagerForm : Form
    {
        public ImageManagerForm()
        {
            InitializeComponent();
        }

        private void LoadImageIDComboBox()
        {
            // load the combo box
            List<int> imageIDList = ProductDB.GetImageIDList();
            foreach (int i in imageIDList)
                imageIDComboBox.Items.Add(i);
        }

        private void ImageManagerForm_Load(object sender, EventArgs e)
        {
            this.LoadImageIDComboBox();
            imageIDComboBox_SelectedIndexChanged(sender, e);
        }

        private void imageIDComboBox_SelectedIndexChanged(
            object sender, EventArgs e)
        {
            try
            {
                int imageID = Convert.ToInt32(imageIDComboBox.Text);

                // read image bytes from the database and display in picture box
                Byte[] imageByteArray = ProductDB.ReadImage(imageID);
                MemoryStream ms = new MemoryStream(imageByteArray);
                imagePictureBox.Image = System.Drawing.Image.FromStream(ms);
                ms.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(this, ex.Message, "Error");
            }               
        }

        private void uploadButton_Click(object sender, EventArgs e)
        {
            try
            {
                int productID = Convert.ToInt32(productIDTextBox.Text);
                string filename = filenameTextBox.Text;
                ProductDB.WriteImage(productID, filename);
                MessageBox.Show(this, "Image upload was successful!",
                    "Upload Confirmation");

                // refresh combo box
                imageIDComboBox.Items.Clear();
                this.LoadImageIDComboBox();

            }
            catch (Exception ex)
            {
                MessageBox.Show(this, ex.Message, "Error");
            }
        }
    }
}

File: SQL Server 2012/Projects/Visual Basic/Chapter 20a/MusicStoreImageManager/ImageManagerForm.vb
Imports System.IO

Public Class ImageManagerForm

    Dim imageIDList As List(Of Integer)

    Private Sub ImageManagerForm_Load(
        ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles MyBase.Load

        ' initialize data if no data exists
        imageIDList = ProductDB.GetImageIDList()
        If imageIDList.Count = 0 Then
            ' upload images
            ProductDB.WriteImage(1, "8601_cover.jpg")
            ProductDB.WriteImage(2, "pf01_cover.jpg")
            ProductDB.WriteImage(3, "jr01_cover.jpg")
        End If

        ' load combo box
        Me.LoadImageIDComboBox()
        imageIDComboBox_SelectedIndexChanged(sender, e)

    End Sub

    Private Sub LoadImageIDComboBox()
        imageIDList = ProductDB.GetImageIDList()
        For Each i As Integer In imageIDList
            imageIDComboBox.Items.Add(i)
        Next

    End Sub

    Private Sub imageIDComboBox_SelectedIndexChanged(
        ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles imageIDComboBox.SelectedIndexChanged

        Try
            Dim imageID As Integer = Convert.ToInt32(imageIDComboBox.Text)

            ' read image bytes from the database and display in picture box
            Dim imageByteArray As Byte() = ProductDB.ReadImage(imageID)
            Dim ms As MemoryStream = New MemoryStream(imageByteArray)
            imagePictureBox.Image = System.Drawing.Image.FromStream(ms)
            ms.Close()
        Catch ex As Exception
            MessageBox.Show(Me, ex.Message, "Error")
        End Try

    End Sub

    Private Sub uploadButton_Click(
        ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles uploadButton.Click

        Try
            Dim productID As Integer = Convert.ToInt32(productIDTextBox.Text)
            Dim filename As String = filenameTextBox.Text
            ProductDB.WriteImage(productID, filename)
            MessageBox.Show(Me, "Image upload was successful!",
                "Upload Confirmation")

            ' refresh combo box
            imageIDComboBox.Items.Clear()
            Me.LoadImageIDComboBox()
        Catch ex As Exception
            MessageBox.Show(Me, ex.Message, "Error")
        End Try

    End Sub

End Class
